sycl/source/detail/device_image_impl.hpp

```diff
@@ -176,7 +182,7 @@ class device_image_impl {
 
   const context &get_context() const noexcept { return MContext; }
 
-  std::vector<kernel_id> &get_kernel_ids_ref() noexcept { return MKernelIDs; }
+  std::shared_ptr<std::vector<kernel_id>> &get_kernel_ids_ref() noexcept { return MKernelIDs; }
```

Comment:
The `ref` part of the name seems a little misplaced with these changed. Maybe it should be renamed to `get_kernel_ids_ptr`?

Reply:
Done.